### PR TITLE
Update to K8S 1.7.3, OVS 2.7, and added GCE automated provisioning and configuration.

### PR DESCRIPTION
**Pull request by bsteciuk**

### CODE REVIEW

**Comment by pires** [comment]
I think you should've split this into multiple, revertible commits (in case you need it in the future).

I also think it's wrong you haven't incorporated #23 commit here and fixed it in a subsequent commit. It doesn't help at all building a community of contributors.

**Comment by bsteciuk** [comment]
@pires Updated PR to include multiple commits. RE: to #23, I agree, and it can still be merged in before mine.  The nature of the demo/automation (pulling the repo on the nodes) led me to incorporate it for testing the changes with my own changes (which include updating the Linux nodes to OVS 2.7), as they went hand in hand.

**Comment by pires** [comment]
One suggestion, you can simply cherry-pick @aserdean commit over here.

**Comment by bsteciuk** [comment]
Commit by @aserdean has been cherry picked into this PR.

File: master/make-certs [hunk 1]
```diff
@@ -33,7 +33,7 @@ openssl x509 -req -in $pem_server_csr -CA $pem_ca -CAkey $pem_ca_key -CAcreatese
 
 # Make server certs accessible to apiserver.
 chgrp $cert_group $pem_ca $pem_ca_key $pem_server $pem_server_key
-chmod 600 $pem_ca_key $pem_server_key
+chmod 660 $pem_ca_key $pem_server_key
```
Comment (pires) [inline master/make-certs]: Private keys should be readable only to their owner. Some applications may even complain about this, like SSH.

File: provisioning/gce/install_k8s.ps1 [hunk 2]
```diff
@@ -0,0 +1,32 @@
+$K8S_PATH="C:\kubernetes"
+$K8S_VERSION="1.7.3"
+$HOSTNAME = hostname
+#$K8S_MASTER_IP = $env:K8S_MASTER_IP
+#;Invoke-RestMethod -URI http://metadata.google.internal/computeMetadata/v1/instance/attributes/apiServer -Headers @{"Metadata-Flavor" = "Google"}
```
Comment (pires) [inline provisioning/gce/install_k8s.ps1]: If one doesn't need the above, I suggest removing the two lines.

**Pull request merged**